### PR TITLE
conditionally remove the facility code before submit

### PR DESCRIPTION
**Pull request by erikphansen**
## Description
Send the facilityCode to the backend. But only send it if the school was selected with the search tool.

TODO:
- [x] Add unit tests if this is a decent approach for this problem.

## Testing done
Tested locally. Confirmed the JSON payload when entering the school manually and when entering it with the search tool. Confirmed that after picking a school with the search tool and then switching to manual entry, the facility code was still not sent.

## Acceptance criteria
- [ ]

## Definition of done
~~- [ ] Events are logged appropriately~~
~~- [ ] Documentation has been updated, if applicable~~
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/edu-benefits/feedback-tool/config/form.js [hunk 1]
```diff
@@ -59,7 +59,11 @@ const {
 } = fullSchema.properties;
 
 const { assistance, programs, school } = educationDetails.properties;
-const { address: schoolAddress, name: schoolName } = school.properties;
+const {
+  address: schoolAddress,
+  name: schoolName,
+  facilityCode,
+} = school.properties;
```
Comment (erikphansen) [inline src/applications/edu-benefits/feedback-tool/config/form.js]: okay, so I guess the drawback of Prettier is that is wrapped this block adding more noise to this particular diff :(

**Comment by erikphansen** [comment]
To be honest, in the case where you have picked a school with the search tool and then decide to manually enter a school, I'm not entirely sure how the form processing on submit reconciles which school info to use. I'm a bit worried that it's a happy accident that it uses the manually entered info :-/ @rianfowler, do you have any insight here?

File: src/applications/edu-benefits/feedback-tool/helpers.js [hunk 2]
```diff
@@ -58,8 +58,32 @@ export function fetchInstitutions({ institutionQuery, page, onDone, onError }) {
...
+// cleared out the (incorrect) facility code will be sent along with the
+// manually entered school address.
+function removeFacilityCodeIfManualEntry(form) {
+  if (
+    get(
+      'data.educationDetails.school.view:searchSchoolSelect.view:manualSchoolEntryChecked',
+      form,
+    )
+  ) {
+    return set(
```
Comment (cehsu) [inline src/applications/edu-benefits/feedback-tool/helpers.js]: We could unset unset here instead.

File: src/applications/edu-benefits/feedback-tool/helpers.js [hunk 3]
```diff
@@ -58,8 +58,32 @@ export function fetchInstitutions({ institutionQuery, page, onDone, onError }) {
   );
 }
 
+// Helper to clear out the facility code. Needed if the code was set via the
+// search tool and then manual address entry was selected. if this is not
+// cleared out the (incorrect) facility code will be sent along with the
+// manually entered school address.
+function removeFacilityCodeIfManualEntry(form) {
```
Comment (cehsu) [inline src/applications/edu-benefits/feedback-tool/helpers.js]: Would it make sense to move this transformation to the BE with the rest of the transformations? @lihanli @jholton

**Review by cehsu — APPROVED** [review]
LGTM. I had a few questions/comments for your consideration.

File: src/applications/edu-benefits/feedback-tool/helpers.js (hunk 2 above)
Comment (erikphansen) [reply src/applications/edu-benefits/feedback-tool/helpers.js]: yessssss... thank you.

File: src/applications/edu-benefits/feedback-tool/helpers.js (hunk 3 above)
Comment (erikphansen) [reply src/applications/edu-benefits/feedback-tool/helpers.js]: Theoretically we could make the `view:manualSchoolEntryChecked` prop a non-`view:` prop so the BE could see it and then also rip that one off as part of the processing. But it makes just as much sense to just do the work here.

File: src/applications/edu-benefits/feedback-tool/config/form.js (hunk 1 above)
Comment (rianfowler) [reply src/applications/edu-benefits/feedback-tool/config/form.js]: net gain

**Review by rianfowler — APPROVED** [review]

**Pull request merged**